### ISSUE TITLE
mypy: error: expected '('  with 2023.8.2

### ISSUE THREAD

**Issue by ogajduse** [KapJI/homeassistant-stubs#349]
I use homeassistant-stubs in the https://github.com/custom-components/feedparser repository.

I have bumped into an issue with the latest version of stubs.
https://github.com/custom-components/feedparser/actions/runs/5876164929/job/15933776158?pr=91#step:4:145

```
 /home/runner/.cache/pre-commit/repoa5dodlk1/py_env-python3.11/lib/python3.11/site-packages/homeassistant-stubs/components/sensor/__init__.pyi:26: error: expected '('  [syntax]
 ```

The issue seems to be in stubs itself.

**Comment by bramstroker** [comment]
I have exactly the same issue.

**Comment by loopj** [comment]
For some reason, a method is being generated with a hyphen in the name (`__mypy-replace`), see https://github.com/KapJI/homeassistant-stubs/blob/main/homeassistant-stubs/components/sensor/__init__.pyi#L26

Still seeing this issue on 2023.8.3

**Comment by bramstroker** [comment]
I did a quick test and can tell the stub is generated correctly with MyPy 1.4.1
When MyPy is upgraded to 1.5.0 or 1.5.1 the extra function is generated causing the syntax error:

```
def __mypy-replace(
```

So it's definitely some regression bug in stubgen code.

Don't have time to create a bug report now at mypy.

Might be an idea for @KapJI to fix MyPy dependency to 1.4.1 for the time being so at least the stubs can be generated correctly now.
As things are completely broken atm.

**Comment by KapJI** [comment]
Thanks for escalating it with mypy team!

**Comment by KapJI** [comment]
Version `2023.11.2` was published which uses mypy 1.7.0. It has multiple improvements for dataclass stubs. Give it a try.
